src/core/unittest/SettingsTest.cpp

```diff
@@ -334,6 +336,277 @@ TEST(SettingsTest, StreamRecvWindowDefaultGetsOverridenByIndividualLimits)
+    ASSERT_EQ(Settings.GreaseQuicBitEnabled, QUIC_DEFAULT_GREASE_QUIC_BIT_ENABLED);
+    ASSERT_EQ(Settings.EcnEnabled, QUIC_DEFAULT_ECN_ENABLED);
+    ASSERT_EQ(Settings.HyStartEnabled, QUIC_DEFAULT_HYSTART_ENABLED);
+    ASSERT_EQ(Settings.EncryptionOffloadAllowed, QUIC_DEFAULT_ENCRYPTION_OFFLOAD_ALLOWED);
+    ASSERT_EQ(Settings.ReliableResetEnabled, QUIC_DEFAULT_RELIABLE_RESET_ENABLED);
+    ASSERT_EQ(Settings.XdpEnabled, QUIC_DEFAULT_XDP_ENABLED);
+    ASSERT_EQ(Settings.QTIPEnabled, QUIC_DEFAULT_QTIP_ENABLED);
+    ASSERT_EQ(Settings.RioEnabled, QUIC_DEFAULT_RIO_ENABLED);
+    ASSERT_EQ(Settings.OneWayDelayEnabled, QUIC_DEFAULT_ONE_WAY_DELAY_ENABLED);
+    ASSERT_EQ(Settings.NetStatsEventEnabled, QUIC_DEFAULT_NET_STATS_EVENT_ENABLED);
+    ASSERT_EQ(Settings.StreamMultiReceiveEnabled, QUIC_DEFAULT_STREAM_MULTI_RECEIVE_ENABLED);
+}
+
+TEST(SettingsTest, QuicSettingsSetDefault_DoesNotOverwriteSetFields)
+{
+    QUIC_SETTINGS_INTERNAL Settings;
+    CxPlatZeroMemory(&Settings, sizeof(Settings));
+
+    // Set a few fields and mark them as set
+    Settings.IsSet.SendBufferingEnabled = 1;
+    Settings.SendBufferingEnabled = 0;
+    Settings.IsSet.PacingEnabled = 1;
+    Settings.PacingEnabled = 0;
+    QuicSettingsSetDefault(&Settings);
+
+    // These should not be overwritten
+    ASSERT_EQ(Settings.SendBufferingEnabled, 0);
+    ASSERT_EQ(Settings.PacingEnabled, 0);
+
+    // But an unset field should be set to default
+    ASSERT_EQ(Settings.MigrationEnabled, QUIC_DEFAULT_MIGRATION_ENABLED);
+}
+
+struct QuicStorageSettingScopeGuard {
+public:
+    QuicStorageSettingScopeGuard(
+        _In_opt_ const char* StorageName,
+        _In_z_ const char* SettingName) :
+        m_StorageName(StorageName), m_SettingName(SettingName) {}
+
+    QuicStorageSettingScopeGuard(const QuicStorageSettingScopeGuard&) = delete;
+    QuicStorageSettingScopeGuard& operator=(const QuicStorageSettingScopeGuard&) = delete;
+
+    QuicStorageSettingScopeGuard(QuicStorageSettingScopeGuard&&) = delete;
+    QuicStorageSettingScopeGuard& operator=(QuicStorageSettingScopeGuard&&) = delete;
+
+    ~QuicStorageSettingScopeGuard() {
+        CXPLAT_STORAGE* Storage;
+        EXPECT_EQ(
+            QUIC_STATUS_SUCCESS,
+            CxPlatStorageOpen(
+                m_StorageName,
+                nullptr,
+                nullptr,
+                CXPLAT_STORAGE_OPEN_FLAG_DELETEABLE|CXPLAT_STORAGE_OPEN_FLAG_WRITABLE,
+                &Storage));
+
+        EXPECT_EQ(
+            QUIC_STATUS_SUCCESS,
+            CxPlatStorageDeleteValue(
+                Storage,
+                m_SettingName));
+
+        CxPlatStorageClose(Storage);
+    }
+
+private:
+    const char* m_StorageName;
+    const char* m_SettingName;
+};
+
+// --- Test: QuicSettingsLoad sets fields from storage ---
+TEST(SettingsTest, QuicSettingsLoad_SetsFieldsFromStorage)
+{
+    CXPLAT_STORAGE* Storage = NULL;
+
+    QUIC_STATUS Status =
+        CxPlatStorageOpen(
+            "TEST",
```

Comment:
nit: I would make this name more explicit (like "MsQuicUnitTestStorage").

Having a unit test that can change persisted state is a touchy subject, since one generally expectsto be able to run unit test safely on a dev device. In case we fail to cleanup the state, we should make where it comes from very obvious.